streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -197,8 +197,8 @@ synchronized boolean lock(final TaskId taskId) throws IOException {
 
         final FileChannel channel;
 
-        try {
-            channel = getOrCreateFileChannel(taskId, lockFile.toPath());
+        try (final FileChannel fileChannel = getOrCreateFileChannel(taskId, lockFile.toPath())) {
```

Comment:
It looks weird to use the resource outside of try block, does this really work?

Reply:
thanks for review 
It works, but it seems strange.
The code was updated